diff --git a/lib/getloadavg.c b/lib/getloadavg.c
index 9da41c16c02..1cb1c01097d 100644
--- a/lib/getloadavg.c
+++ b/lib/getloadavg.c
@@ -499,7 +499,8 @@ getloadavg (double loadavg[], int nelem)
   }
 # endif
 
-# if !defined (LDAV_DONE) && (defined __linux__ || defined __ANDROID__)
+# if !defined (LDAV_DONE) && (defined __linux__ || defined __ANDROID__) \
+  && (!defined __ANDROID__ || __ANDROID_API__ >= 13)
                                       /* Linux without glibc, Android, Cygwin */
 #  define LDAV_DONE
 #  undef LOAD_AVE_TYPE
